scripts/backupmanager.sh

```diff
@@ -0,0 +1,336 @@
+        exit 1
+    fi
+
+    # Check if backup directory exists
+    if [ ! -d "$BACKUP_DIR" ]; then
+        pp --error "[ERROR] Backup directory $BACKUP_DIR does not exist.\n"
+        exit 1
+    fi
+
+    # Check if there are any files in the backup directory
+    if [ -z "$(ls -A "$BACKUP_DIR")" ]; then
+        pp --info "No backups in the backup directory $BACKUP_DIR.\n"
+        exit 0
+    fi
+
+    # get file list
+    files=$(find "$BACKUP_DIR" -maxdepth 1 -type f -name "*.tar.gz" | sort -r)
+    total_file_count=$(echo "$files" | wc -l)
+
+    if [ -z "$num_backup_entries" ]; then
+        # if num_backup_entries is not set, use all files
+        file_list=$files
+    else
+        # if num_backup_entries is set, use the first num_backup_entries files
+        file_list=$(echo "$files" | head -n "$num_backup_entries")
+    fi
+
+    # print file list (currently using date from file name but can use creation date from 'stat')
+    for file in $file_list; do
+        filename=$(basename "$file")
+        # get date from creation date
+        #creation_date=$(stat -c %w "$file")
+
+        # Reformat the date string
+        #date=$(date -d "$creation_date" +'%Y-%m-%d %H:%M:%S')
+
+        # get date from filename
+        date_str=${filename#saved-}    # Remove 'saved- or restore-' prefix
+        date_str=${date_str%.tar.gz}   # Remove '.tar.gz' suffix
+
+        # Reformat the date string
+        date=$(date -d "${date_str:0:8} ${date_str:9:2}:${date_str:11:2}:${date_str:13:2}" +'%Y-%m-%d %H:%M:%S')
+
+        pp --info "${date} | "
+        pp --base "${filename}\n"
+    done
+
+    if [ -z "$num_backup_entries" ]; then
+        # if num_backup_entries is not set, use all files
+        pp --info "\n> Found ${total_file_count} backup file(s)!\n"
+    else
+        # if num_backup_entries is set, use the first num_backup_entries files
+        pp --info "\n> Found ${total_file_count} backup file(s), but listing only ${num_backup_entries} file(s)!\n"
+    fi
+}
+
+function clean_backups() {
+    local num_files_to_keep=${1:-${BACKUPS_TO_KEEP}}
+
+    pp --success ">>> Backup cleaning started..\n\n"
+
+    if [[ -z "$num_files_to_keep" ]]; then
+        pp --warning "> Number of backups to keep is empty. Using default value of 30.\n"
+        num_files_to_keep=30
+    fi
+
+    if ! [[ "$num_files_to_keep" =~ ^[0-9]+$ ]]; then
+        pp --error "> [ERROR] Invalid argument '${BACKUPS_TO_KEEP}'. Please provide a positive integer.\n\n"
+        print_usage
+        exit 1
+    fi
+
+    # Check if there are any files in the backup directory
+    if [ -z "$(ls -A "${BACKUP_DIR}")" ]; then
+        pp --info "> No files in the backup directory ${BACKUP_DIR}. Exiting...\n\n"
+        exit 0
+    fi
+
+    pp --info "> Keeping latest ${num_files_to_keep} backups.\n"
+    
+    files_to_delete=$(find "${BACKUP_DIR}" -maxdepth 1 -type f -name "saved-*.tar.gz" \
```

Comment:
```
ls -1t saved-*.tar.gz | tail -n +$(($BACKUP_RETENTION_AMOUNT_TO_KEEP + 1)) | xargs -d '\n' rm -f --
```
Does that same but faster and in 1 line instead of 10-15

Not every old code i had is bad you know?